5-20/5-20.md

```diff
@@ -127,11 +113,14 @@ cc := 3.14159 float64 [3.14159]
 dd := true    bool [true]
 ```
 
-##### Conversion vs casting
+##### 변환과 타입 변경(Conversion vs casting)
 
 Go doesn't have casting, but conversion. Instead of telling a compiler to pretend
 to have some more bytes, we have to allocate more memory like so.
 
+Go 는 `casting` 을 지원하지 않고 `conversion` 을 지원한다. 컴파일러에게 몇 바이트 더 있는 척
+하기보다 실제로 메모리를 더 할당한다.
```

Comment:
어색한데 코드도 어떻게 연관되는지 몰라서 뭐라고 해야 될지 모르겠네요. 다른 분들도 봐주세요.

Reply:
덕분에 좀더 챙겨보았습니다. 아래의 링크가 답이 되겠습니다.
https://www.geeksforgeeks.org/type-casting-or-type-conversion-in-golang/

`aaa := 10` 이라고 해도 만약 32-bit os 라면 4바이트 크기의 메모리를 사용할 것입니다. 즉, int 와 int32 는 정확히 같은 메모리 사이즈를  사용합니다. 하지만, 그럼에도 불구하고 Go 언어에서는 int32(10) 이라고 `명확히` 명시해두어야 int와 구분되는 int32 타입으로 저장합니다. 

예시가 좀더 선명했으면 좋았을 것 같기는 합니다. 최소한 아래 정도의 예시? 

```go
var a int32
b := 10
a = int32(b) 
```

Reply:
메모리에 관한 부분은, Go의 경우는 예를 들어 int16 인걸 int64 에 대입해도 당장 메모리를 64비트 할당하지 않고도 마치 int64 인척 챙겨주지 않는다는 의미로 읽었습니다.

Reply:
😢 아직도 이해하지 못하였습니다. 명시적 변환은 이해했습니다.

> 컴파일러에게 몇 바이트 더 있는 척하기보다 실제로 메모리를 더 할당한다.

이게 이해가 가지 않네요. 몇 바이트 더 있는 척 한다는건 무슨 말이고 실제로 더 할당한다는 건 무슨 뜻인지

Reply:
말씀대로 좀더 들여다 봐야겠습니다. 질문 주셔서 덕분에 좀더 고민하게 되네요.

Reply:
아~ 이해가 될 것 같습니다. (아래는 일종의 sudo code)
x 가 int 이고 y 가 float, z 가 int 라 할때에 
z = x + y 라고 하면 

`몇 바이트 더 있는 척` 하는 경우에는 rvalue 의 x + y 에서 x 를 임시로 float 로 바꿔서 x + y 를 계산합니다. 하지만 실제로 x라는 변수가 자리잡은 메모리에는 변하는 것이 없습니다.

`하지만 Go 처럼 실제로 변하는 경우에는` 실제로 x 가 위치하는 메모리쪽에 변화가 일어난다. 

이정도가 아닐까 싶네요. 대략의 예상이고 실제로 테스트로 검증한 것은 아닙니다만. 
원저자에게 문의해보겠습니다.

Reply:
int랑 float 이 사이즈가 다르다구 한것도 이상하네요.
일반적으로 둘다 4바이트 차지하니까요.

물어봐주시면 감사하겠습니다.

Reply:
아래는 Hoan 의 회신입니다. 제가 말한 내용과 비슷하기는 합니다만... 좀 마음에 안드네요. 아예 빼버렸으면 싶습니다. 이정도 깊이까지는 알 필요도 없다 싶구요.

```
Let’s say you have 2 variables that have the same types, the compiler let you assign one to another. However, if you have 2 variables that have different types, then you can’t just assign one to another. The compiler wouldn't like that. Now, in many languages, you can do that and what happens behind the scene is that the compiler pretends or adds more bytes to one of these 2 variables so that they share the same size and structure. However, you can’t do it in Go. You have to explicitly convert one to the other type to be able to assign them together.
```